olive/passes/onnx/transformer_optimization.py

```diff
@@ -94,30 +94,62 @@ def sd_model_types():
         import onnxruntime as ort
 
-        input_model_type = run_config["model_type"]
+        ort_version = version.parse(ort.__version__)
+        is_ort_1_13_or_older = ort_version < version.parse("1.14.0")
+        # is_ort_1_14 = ort_version >= version.parse("1.14.0") and ort_version < version.parse("1.15.0")
 
         # default to no specific fusion options in earlier releases of ORT
-        if version.parse(ort.__version__) < version.parse("1.14.0"):
+        if is_ort_1_13_or_older:
             return
 
-        from onnxruntime.transformers.fusion_options import FusionOptions
+        is_ort_1_15_0_or_newer = ort_version >= version.parse("1.15.0")
+        is_ort_1_15_1_or_newer = ort_version >= version.parse("1.15.1")
 
-        is_ort_115_or_newer = version.parse(ort.__version__) >= version.parse("1.15.0")
-        if not is_ort_115_or_newer and input_model_type != "unet":
+        input_model_type = run_config["model_type"]
+        if not is_ort_1_15_0_or_newer and input_model_type != "unet":
             # 'vae' and 'clip' are only recognized in ORT v1.15+. earlier versions of ORT stable diffusion
             # optimization simply use "unet" for these model types.
             run_config["model_type"] = "unet"
 
+        from onnxruntime.transformers.fusion_options import FusionOptions
+
         fusion_options = FusionOptions(run_config["model_type"])
 
-        if input_model_type == "unet":
-            fusion_options.enable_packed_kv = use_float16
-            fusion_options.enable_packed_qkv = use_float16 and is_ort_115_or_newer
-            fusion_options.enable_bias_add = is_ort_115_or_newer
+        # TODO: remove dml_future when ORT 1.15.1 with future version of DML is released
+        # This "provider" value is simply a way to test in-development changes without having
+        # to bump the ORT version.
+        dml_future = pass_config["target_provider"] == "directml_future"
+
+        if pass_config["target_provider"] == "directml" or dml_future:
+            # Some of these fusions are disabled because they provide no performance advantage,
+            # and it's preferable to limit ops outside the ONNX domain.
+            fusion_options.enable_gelu = is_ort_1_15_0_or_newer
+            fusion_options.enable_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_attention = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.use_multi_head_attention = is_ort_1_15_1_or_newer or dml_future
+            fusion_options.enable_skip_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_embed_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_bias_skip_layer_norm = is_ort_1_15_0_or_newer
+            fusion_options.enable_bias_gelu = is_ort_1_15_0_or_newer
+            fusion_options.enable_gelu_approximation = False
+            fusion_options.enable_qordered_matmul = is_ort_1_15_0_or_newer
```

Comment:
`QOrderedMatMul` I don't see any registrations for it (in the non-Pat branches anyway), and so propose setting false. I wouldn't substitute to any operators that we don't actually have DML registrations for in OperatorRegistration.cpp, because otherwise we *could* get *worse* perf if the CPU registered that contrib op but DML did not.

Also as a general compat/interchange principle, I'd favor emitting as close to standard ONNX as possible, *except* in cases where we know perf is improved by it. The higher the contrib op ratio in a model, the less we can rationalize about expected behavior (the contrib ops are not as well documented, do not have the same breadth/history of testing, and are not opset versioned), and the fewer user tools which can operate with them.